websockets.md

```diff
@@ -0,0 +1,222 @@
+---
+title: WebSockets
+---
+
+The Common Lisp ecosystem boasts a few approaches to building WebSocket servers.
+First, there is the excellent
+[Hunchensocket](https://github.com/joaotavora/hunchensocket) that is written as
+an extension to [Hunchentoot](https://edicl.github.io/hunchentoot/), the classic
+web server for Common Lisp.  I have used both and I find them to be wonderful.
+
+Today, however, you will be using the equally excellent
+[websocket-driver](https://github.com/fukamachi/websocket-driver) for use with
+[Clack](https://github.com/fukamachi/clack). The community has expressed a
+slight prefernce for the Clack eco system, as it provides a uniform interface to
+a variety of backends, including Hunchentoot. You can pick and choose the
+backend.
+
+In what follows, you will build a one-room chat server and connect to it from a
+web browser. The tutorial is written so that you can enter the code into your
+REPL as you go. The full code listing is repeated on the end.
+
+As a first step, you should load the needed libraries via quicklisp:
+
+~~~lisp
+
+(ql:quickload '(clack websocket-driver alexandria))
+
+~~~
+
+
+## The websocket-driver Concept
+
+In websocket-driver, a WebSocket connection is an instance of the `ws` class,
+which exposes an event-driven API. You register event handlers by passing your
+WebSocket instance the method `on`, e.g. `(on :message my-websocket #'some-message-handler)`, 
+where `some-message-handler` would be evoked whenever a new message arrives.
+
+The `websocket-driver` API provides for handlers on the following events:
+
+- `:open`: When a connection is opened. Expects a thunk as its handler.
```

Comment:
What is a thunk? Maybe you should replace that word?

Reply:
Ah sure. Sorry a thunk is a function of zero arguments. I can just say that.